### PR TITLE
Updated code snippet question post API

### PR DESCRIPTION
**Pull request by Deepankar-promact**
**Fixed Issues**

#21028

**Files Changed**

CodeSnippetQuestionAC.cs - Renamed CodeSnippetQuestionDto to CodeSnippetQuestionAC
QuestionAC.cs - Changed CodeSnippetQuestion to CodeSnippetQuestionAC
IQuestionRepository.cs - Changed return type of function AddCodeSnippetQuestion to Task
QuestionRepository.cs - Updated function AddCodeSnippetQuestion according to new model
Startup.cs - Added configuration to ignore self referencing loop

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [ ] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [ ] All test cases are executed provided by tester.

**Known Issues**

-no known issue

**Comments**

No work done at client side so "client side code comments" is unchecked.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 1]
```diff
@@ -44,29 +47,47 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
...
             {
-                var question = _dbContext.CodeSnippetQuestion.Add(codeSnippetQuestion);
-                _dbContext.SaveChanges();
-                var codingLanguageList = codeSnippetQuestionModel.LanguageList;
-                //To-Do Change according to new model var questionId = question.Entity.Id; 
+                //Add common question details
+                var question = await _dbContext.Question.AddAsync(questionAC.Question);
+                await _dbContext.SaveChangesAsync();
+
+                var questionId = question.Entity.Id;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: No need to take extra variable.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 2]
```diff
@@ -44,29 +47,47 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
...
+            CodeSnippetQuestion codeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC, CodeSnippetQuestion>(codeSnippetQuestionModel);
+
             using (var transaction = _dbContext.Database.BeginTransaction())
             {
-                var question = _dbContext.CodeSnippetQuestion.Add(codeSnippetQuestion);
-                _dbContext.SaveChanges();
-                var codingLanguageList = codeSnippetQuestionModel.LanguageList;
-                //To-Do Change according to new model var questionId = question.Entity.Id; 
+                //Add common question details
+                var question = await _dbContext.Question.AddAsync(questionAC.Question);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: no extra variable required

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 3]
```diff
@@ -44,29 +47,47 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
...
+                await _dbContext.SaveChangesAsync();
+
+                var questionId = question.Entity.Id;
+
+                //Add codeSnippet part of question
+                codeSnippetQuestion.QuestionId = questionId;
+                var codingQuestion = await _dbContext.CodeSnippetQuestion.AddAsync(codeSnippetQuestion);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: no extra variable required

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 4]
```diff
@@ -44,29 +47,47 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
...
+                await _dbContext.SaveChangesAsync();
+
+                var codingQuestionId = codingQuestion.Entity.Id;
+                var codingLanguageList = codeSnippetQuestionModel.LanguageList; 
+
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: optimize code

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs (hunk 2 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: question variable is used to get Id generated after saveChanges.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs (hunk 1 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: I will remove this variable.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs (hunk 3 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: codingQuestion is used to get Id generated after saveChanges in CodeSnippetQuestion table.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 5]
```diff
@@ -44,29 +47,42 @@ public ICollection<SingleMultipleAnswerQuestionApplicationClass> GetAllQuestions
...
             using (var transaction = _dbContext.Database.BeginTransaction())
             {
-                var question = _dbContext.CodeSnippetQuestion.Add(codeSnippetQuestion);
-                _dbContext.SaveChanges();
-                var codingLanguageList = codeSnippetQuestionModel.LanguageList;
-                //To-Do Change according to new model var questionId = question.Entity.Id; 
-                foreach (var language in codingLanguageList)
+                //Add common question details
+                var question = await _dbContext.Question.AddAsync(questionAC.Question);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: please optimize code

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: please optimize code

**Pull request merged**